Support for YAML configuration

theolind opened the pull request:
Added support for yaml configuration files. Additionally main now perfers .yaml over .conf and can create a default .yaml file if no configuration files are found.

andythigpen committed:
Add configurable intervals to device tracker.

Allows the user to configure a longer interval for when the device
tracker should scan for devices.

balloob committed:
Merge pull request #36 from andythigpen/feature/device-tracker-interval

Add configurable intervals to device tracker.

balloob committed:
Merge branch 'dev'

* dev:
  Device tracker: make all Mac addresses uppercase. Fixes #37
  ps: Slim down README.md
  Update __init__.py
  Remove no longer supported options from example
  Adds event automation module.

balloob committed:
Merge branch 'dev'

* dev:
  Fix device tracker tests

theolind committed:
Added support for YAML configuration files

theolind committed:
Added support for YAML files, now with included python modules

theolind committed:
Added logic to main that prefers .yaml-config over .conf

theolind commented:
Don't know what happened with all the commits there, sorry. Maybe I should have branched off from the dev branch and not the master branch. Shall I give it another try?

balloob commented:
It's fine. Those changes won't change anything. The CI is failing because I accidently pushed a style violation to dev :cry: Testing it locally atm and will merge if it works. Thanks !

balloob commented:
Oh, I was wrong, there is still 1 style violation:

```
************* Module homeassistant.bootstrap
C:117, 0: Unnecessary parens after 'if' keyword (superfluous-parens)
```

Also, could you remove home-assistant.conf.example and create home-assistant.conf.yaml?

I'll work on updating home-assistant.io today.

theolind committed:
Added example yaml config file, removed old example config, fixed style violation in bootstrap.py

theolind commented:
I have added an example yaml config now and removed the old one. The only thing that does not seem to work is the custom component example.py. A few modules are bit troublesome for me to test since I don't own any of the devices.

balloob commented:
The example component is not working because yaml transforms the following:

```yaml
example:
```

into

```python
{
  "example": None
}
```

Where components expect the value to be a dictionary instead of `None`.

This can be fixed by replacing [lines 80-82 in bootstrap.py](https://github.com/theolind/home-assistant/blob/feature_yaml_config/homeassistant/bootstrap.py#L80) with:

```python
    # Make a copy because we are mutating it.
    # Convert it to defaultdict so components can always have config dict
    # Convert values to dictionaries if they are None
    config = defaultdict(
        dict, {key: value or {} for key, value in config.items()})
```

When that's done it's ready for merging :smile:

balloob commented:
To help users migrate we'll need to write a blog post summarizing the changes. If you like it you can do it. I have no problems doing it myself.

Writing a blog post is as easy as cloning [the website repo](https://github.com/balloob/home-assistant.io) and adding a new file to [the _posts folder](https://github.com/balloob/home-assistant.io/tree/master/source/_posts).

theolind committed:
Now converting config values to dictionaries if they are None

theolind commented:
Sure, I'll take care of a blog post!

theolind committed:
Fixed style error, a line was too long

balloob commented:
Merged it, thanks !

For your blog post, add `author:` to the YAML on the top to have it display your name as author. Make a pull request back to my branch and we can push it out.
